internal/kafka/internal/config/kafka_supported_instance_types.go

```diff
@@ -0,0 +1,202 @@
+	if kp.Id == "" || kp.DisplayName == "" || len(kp.Sizes) == 0 {
+		return fmt.Errorf("Kafka instance type '%s' is missing required parameters.", kp.Id)
+	}
+
+	if !arrays.Contains(types.ValidKafkaInstanceTypes, kp.Id) {
+		return fmt.Errorf("kafka instance type id '%s' is not valid. Valid kafka instance types are: '%v'", kp.Id, types.ValidKafkaInstanceTypes)
+	}
+
+	existingSizes := make(map[string]int, len(kp.Sizes))
+
+	for _, kafkaInstanceSize := range kp.Sizes {
+		if _, ok := existingSizes[kafkaInstanceSize.Id]; ok {
+			return fmt.Errorf("Kafka instance size '%s' for instance type '%s' was defined more than once.", kafkaInstanceSize.Id, kp.Id)
+		}
+		existingSizes[kafkaInstanceSize.Id]++
+
+		if err := kafkaInstanceSize.validate(kp.Id); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+type KafkaInstanceSize struct {
+	Id                          string   `yaml:"id"`
+	IngressThroughputPerSec     Quantity `yaml:"ingressThroughputPerSec"`
+	EgressThroughputPerSec      Quantity `yaml:"egressThroughputPerSec"`
+	TotalMaxConnections         int      `yaml:"totalMaxConnections"`
+	MaxDataRetentionSize        Quantity `yaml:"maxDataRetentionSize"`
+	MaxPartitions               int      `yaml:"maxPartitions"`
+	MaxDataRetentionPeriod      string   `yaml:"maxDataRetentionPeriod"`
+	MaxConnectionAttemptsPerSec int      `yaml:"maxConnectionAttemptsPerSec"`
+	QuotaConsumed               int      `yaml:"quotaConsumed"`
+	QuotaType                   string   `yaml:"quotaType"`
+	CapacityConsumed            int      `yaml:"capacityConsumed"`
+}
+
+// validates Kafka instance size configuration to ensure the following:
+//
+// - all properties must be defined
+// - any non-id string values must be parseable
+// - any int values must not be less than or equal to zero
+func (k *KafkaInstanceSize) validate(instanceTypeId string) error {
+	if k.EgressThroughputPerSec.IsEmpty() || k.IngressThroughputPerSec.IsEmpty() ||
+		k.MaxDataRetentionPeriod == "" || k.MaxDataRetentionSize.IsEmpty() || k.Id == "" || k.QuotaType == "" {
+		return fmt.Errorf("Kafka instance size '%s' for instance type '%s' is missing required parameters.", k.Id, instanceTypeId)
+	}
+
+	egressThroughputQuantity, err := k.EgressThroughputPerSec.ToK8Quantity()
+	if err != nil {
+		return fmt.Errorf("egressThroughputPerSec for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	ingressThroughputQuantity, err := k.IngressThroughputPerSec.ToK8Quantity()
+	if err != nil {
+		return fmt.Errorf("ingressThroughputPerSec for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	maxDataRetentionSize, err := k.MaxDataRetentionSize.ToK8Quantity()
+	if err != nil {
+		return fmt.Errorf("maxDataRetentionSize for Kafka instance type '%s', size '%s' is invalid: %s", k.Id, instanceTypeId, err.Error())
+	}
+
+	maxDataRetentionPeriod, err := duration.ParseISO8601(k.MaxDataRetentionPeriod)
```

Comment:
Assuming this parse function only validates the format, should there be a check to determine if the provided time is less than current time? meaning a past time than current time cannot be valid.

Reply:
@Rajagopalan-Ranganathan this is a duration (e.g. `P14D` for 14 days), not a timestamp. So this can't lay in the past. Or did you mean something different?

Reply:
Then its perfect, i assumed it was a timestamp - perfect!